tests/tracking/test_tracking.py

```diff
@@ -34,6 +31,33 @@ def test_create_experiment():
         tracking.set_tracking_uri(None)
 
 
+def test_set_experiment():
+    with pytest.raises(TypeError):
+        mlflow.set_experiment()
+
+    with pytest.raises(Exception):
+        mlflow.set_experiment(None)
+
+    with pytest.raises(Exception):
+        mlflow.set_experiment("")
+
+    assert not tracking.fluent._active_experiment_id
+
+    try:
+        with TempDir() as tracking_uri:
+            tracking.set_tracking_uri(tracking_uri.path())
+            name = "random_exp"
+            exp_id = mlflow.create_experiment(name)
+            mlflow.set_experiment(name)
+            assert mlflow.tracking.fluent._active_experiment_id == exp_id
```

Comment:
In this case, we are only testing that `_active_experiment_id` is set, but we do not test that runs actually use `_active_experiment_id`. Could we try creating a run and confirming that the experiment_id is instead set to the appropriate thing? Then we could stop testing the internal implementation detail in `_active_experiment_id` altogether.

Reply:
sweet! I like that.